src/sentry/profiles/consumers/process/factory.py

```diff
@@ -1,19 +1,29 @@
 from typing import Mapping
 
 from arroyo.backends.kafka.consumer import KafkaPayload
-from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.abstract import (
+    MessageRejected,
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
 from arroyo.processing.strategies.commit import CommitOffsets
 from arroyo.processing.strategies.run_task import RunTask
 from arroyo.types import Commit, Message, Partition
 
+from sentry.monitoring.queues import is_queue_healthy, monitor_queues
 from sentry.profiles.task import process_profile_task
 
 
 def process_message(message: Message[KafkaPayload]) -> None:
+    if not is_queue_healthy("profiles.process"):
```

Comment:
IMO it is fair to check redis for every single event, but this adds up once you check multiple queues with different keys if you wanna do that. Not sure if it makes sense to just have a single atomic bool that you can write to in your monitoring thread, instead of going the route to redis.

Reply:
Yeah this is definitely something to be ironed out.